### PR TITLE
Feature: Implement and Document New Interactive Mode Commands

### PR DESCRIPTION
**Pull request by moming2k**
**Description:**

This PR introduces new commands for the interactive mode and updates the `README.md` to document these commands. The goal is to enhance the user experience by providing new functionalities and ensuring users are well-informed about how to use them.

**Changes:**
- Implemented new commands in the interactive mode:
  - `%debug [true/false]`: Toggle debug mode.
  - `%reset`: Reset the current session.
  - `%save_message [path]`: Save messages to a specified JSON path.
  - `%load_message [path]`: Load messages from a specified JSON path.
  - `%help`: Display the help message.
- Updated the `README.md`:
  - Added a section titled "Interactive Mode Commands".
  - Listed and described all the new commands.

**Impact:**
This update will provide users with new functionalities in the interactive mode and ensure they have clear documentation to refer to. This combination aims to improve user experience and interaction with the tool.

**Request for Review:**
Kindly review the changes in both the codebase and the documentation. Feedback on the implementation, command usability, and documentation clarity would be greatly appreciated. If there are suggestions for additional commands or improvements, please let me know.

### CODE REVIEW

File: interpreter/interpreter.py [hunk 1]
```diff
@@ -180,27 +180,111 @@ def get_info_for_system_message(self):
...
-
     # ^ verify_api_key may set self.local to True, so we run this as an 'if', not 'elif':
     if self.local:
 
       # Code-Llama
       if self.llama_instance == None:
 
-        # Find or install Code-Llama
-        try:
+        # Find or install Code-Llama        try:
```
Comment (moming2k) [inline interpreter/interpreter.py:287]:
```suggestion
        # Find or install Code-Llama
        try:
```

File: interpreter/interpreter.py [hunk 2]
```diff
@@ -280,31 +365,29 @@ def chat(self, message=None, return_messages=False):
...
         readline.add_history(user_input)
 
+        if user_input.startswith("%") or user_input.startswith("/"):
+          self.handle_command(user_input)
+          continue
+
         # Add the user message to self.messages
         self.messages.append({"role": "user", "content": user_input})
 
         # Let the user turn on debug mode mid-chat
```
Comment (moming2k) [inline interpreter/interpreter.py:375]:
```suggestion

```

File: interpreter/interpreter.py [hunk 3]
```diff
@@ -319,8 +402,7 @@ def verify_api_key(self):
         self.azure_deployment_name = os.environ['AZURE_DEPLOYMENT_NAME']
         self.azure_api_type = os.environ.get('AZURE_API_TYPE', 'azure')
       else:
-        # This is probably their first time here!
-        self._print_welcome_message()
+        # This is probably their first time here!        self._print_welcome_message()
```
Comment (moming2k) [inline interpreter/interpreter.py:405]:
```suggestion
        # This is probably their first time here!
        self._print_welcome_message()
```

**Comment by moming2k** [comment]
cleared all the whitespace problem, ready for review

**Comment by KillianLucas** [comment]
Wow @moming2k — This is something else. Packs a punch of functionality that a **lot** of people have been wanting (myself included) implemented it very simply and clearly. Great PR. Merging now.

**Pull request merged**